core/common-util/src/main/java/datawave/core/cache/LRUMapClassCache.java

```diff
@@ -0,0 +1,37 @@
+package datawave.core.cache;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.commons.collections4.map.LRUMap;
+
+/**
+ * A {@link LRUMap} implementation of a {@link ClassCache}
+ */
+public class LRUMapClassCache implements ClassCache {
+
+    private final Map<String,Class<?>> cache;
+
+    public LRUMapClassCache() {
+        this(DEFAULT_SIZE);
+    }
+
+    public LRUMapClassCache(int size) {
+        cache = Collections.synchronizedMap(new LRUMap<>(size));
+    }
+
+    @Override
+    public String name() {
+        return "LRUMap";
+    }
+
+    @Override
+    public Class<?> get(String name) throws ClassNotFoundException {
+        Class<?> clazz = cache.get(name);
```

Comment:
Not a big deal but given that we're calling get + set separately, there could be a race condition here. It seems like that's probably fine, we just call the `Class.forName()` twice.

Reply:
Ideally we never use this cache, it's here to act as a reference point for the other two implementations. Agree that it's probably fine.